tests/queries/0_stateless/02956_fix_to_start_of_milli_microsecond.sql

```diff
@@ -0,0 +1,4 @@
+SELECT toStartOfInterval(toDateTime64('2023-10-09 10:11:12.000999', 6), toIntervalMillisecond(1));
```

Comment:
Isn't it generally a strange behavior that a **toStartOfSomething** function returns a value that is _past_ the given interval? 

I.e. here It is not **toStartOfMillisecond** anymore. It is more of a **somewhereNearTheGivenMillesecondAndMaybeEvenInFuture** :)